typing/typeopt.ml

```diff
@@ -155,6 +155,9 @@ let classify ~classify_product env loc ty sort : _ classification =
   | Tconstr (p, _args, _abbrev) ->
       if Path.same p Predef.path_float then Float
       else if Path.same p Predef.path_lazy_t then Lazy
+      else if Path.same p Predef.path_int
+           || Path.same p Predef.path_int8
+           || Path.same p Predef.path_int16 then Int
```

Comment:
Why add those lines? Shouldn't those be always gc-ignorable and handled by the case above?